packages/typespec-client-generator-core/src/http.ts

```diff
@@ -569,6 +534,50 @@ export function getCorrespondingMethodParams(
   return diagnostics.wrap([]);
 }
 
+function findMapping(
+  methodParameters: SdkModelPropertyType[],
+  serviceParam: SdkHttpParameter | SdkModelPropertyType
+): SdkModelPropertyType | undefined {
+  const queue: SdkModelPropertyType[] = [...methodParameters];
+  const visited: Set<SdkModelType> = new Set();
+  while (queue.length > 0) {
+    const methodParam = queue.shift()!;
+    if (
+      methodParam.__raw &&
+      serviceParam.__raw &&
+      (methodParam.__raw === serviceParam.__raw ||
+        methodParam.__raw === serviceParam.__raw.sourceProperty)
+    ) {
+      return methodParam;
+    }
+    if (
+      serviceParam.kind === "header" &&
+      serviceParam.serializedName === "Content-Type" &&
+      methodParam.name === "contentType"
+    ) {
+      return methodParam;
+    }
+    if (
+      serviceParam.kind === "header" &&
+      serviceParam.serializedName === "Accept" &&
+      methodParam.name === "accept"
+    ) {
+      return methodParam;
+    }
```

Comment:
This appears to be somewhat hack? Is it possible for us to know they are all?

PS: for header, should we always do case-insensitive on serializedName (path/query on serializedName  as well)? That would cover the "accept".

Reply:
Fine. I am not sure about the logic. Just from the name, if serviceOperation does not have such param, why there is `serviceParam` of such. They are also added?

Reply:
I would also agree with always do case-insensitive for serializedName, I don't think we need to keep both lowercase name and first-letter capital case serializedName.

Reply:
OK. May worth a comment as well. Next dev may not familiar with this.